flow/layers/backdrop_filter_layer.cc

```diff
@@ -11,9 +11,21 @@ BackdropFilterLayer::BackdropFilterLayer(sk_sp<SkImageFilter> filter)
 
 void BackdropFilterLayer::Preroll(PrerollContext* context,
                                   const SkMatrix& matrix) {
+  size_t count = 0;
+  if (context->damage_context) {
+    count = context->damage_context->layer_entries_count();
+  }
   Layer::AutoPrerollSaveLayerState save =
       Layer::AutoPrerollSaveLayerState::Create(context, true, bool(filter_));
   ContainerLayer::Preroll(context, matrix);
+  if (context->damage_context) {
+    // Adjust display bounds to layers below this layer
+    context->damage_context->ApplyImageFilter(0, count, filter_.get(), matrix,
+                                              paint_bounds());
+    // Push entry before children
+    context->damage_context->PushLayerContribution(this, compare, matrix, *context,
+                                            count);
```

Comment:
Why does this need to be pushed after the children are prerolled?

Reply:
It needs to be done after children are prerolled, because otherwise we don't know the paint bounds.

Reply:
Can it be pushed and then have its bounds updated later? Something like:

```
    Contribution c = push(me);
    ...recurse...;
    c.adjustBounds();
```

Reply:
Sure. I wanted to make LayerContribution private but I don't have any strong feelings about it. Pushing it on arbitrary index does feel bit weird, so this is probably better.

Reply:
If it only needs to adjust the bounds, then it could be private and hidden behind a more public interface that exposes just that functionality.